plugins/woocommerce/src/Blocks/Shipping/ShippingController.php

```diff
@@ -129,16 +129,40 @@ public function show_local_pickup_details( $return_value, $order ) {
 		$details         = $shipping_method->get_meta( 'pickup_details' );
 		$location        = $shipping_method->get_meta( 'pickup_location' );
 		$address         = $shipping_method->get_meta( 'pickup_address' );
+		$cost            = $shipping_method->get_total();
 
-		if ( ! $address ) {
+		$section = '';
+
+		if ( $location ) {
+			$section .= sprintf(
+				// Translators: %s location name.
+				__( 'Collection from <strong>%s</strong>:', 'woocommerce' ),
+				$location
+			);
+		}
+
+		if ( $address ) {
+			$section .= '<br/><address>' . nl2br( esc_html( str_replace( ',', ', ', $address ) ) ) . '</address>';
+		}
+
+		if ( $details ) {
+			$section .= '<br/>' . wp_kses_post( $details );
+		}
+
+		if ( $cost > 0 ) {
+			$section .= '<br/>' . sprintf(
+				// Translators: %s is the formatted price.
+				__( 'Pickup cost: %s', 'woocommerce' ),
+				wc_price( $cost, array( 'currency' => $order->get_currency() ) )
```

Comment:
This doesn't deal with prices inclusive or exclusive of tax like the `get_shipping_to_display` method does. What if we used the original `$return_value` instead of regenerating it? Either that or we need to account for the logic in the `get_shipping_to_display` method.